cmd/cli/mesh_upgrade.go

```diff
@@ -37,9 +30,13 @@ overridden from the default for the "osm install" command, the --mesh-name and
 By default, if flags tied to Helm chart values are not specified for "osm mesh
 upgrade", then the value from the current release will be carried over to the
 new release. Two exceptions to this rule are --container-registry and
---osm-image-tag, which will be overridden from the old release by default. Note
-that edits to resources NOT made by Helm or the OSM CLI may not persist after
-"osm mesh upgrade" is run.
```

Comment:
This note is still relevant if edits are made to other resources with `kubectl edit` or similarly, so I would keep it here.

Reply:
I added the note back